File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
int reverse(int x) {
    int ans = 0;
    while (x != 0) {
        int rem = x % 10;
        x /= 10;

        // Check for overflow before it happens
        if (ans > INT_MAX / 10 || (ans == INT_MAX / 10 && rem > 7)) return 0; // Overflow for positive numbers
        if (ans < INT_MIN / 10 || (ans == INT_MIN / 10 && rem < -8)) return 0; // Overflow for negative numbers

        ans = ans * 10 + rem;
    }
    return ans;
}
};

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        reversednum = 0
        sign = -1 if(x<0) else +1
        a = abs(x)
        while(a!=0):
            num = a%10
            reversednum = reversednum*10 + num
            a= a//10
        
        reversednum *= sign

        if (reversednum<(-1<<31) or reversednum>((1<<31))-1):
            return 0

        return reversednum 

